File: backend/src/main/java/com/example/backend/entity/Ingredient.java
package com.example.backend.entity;

import lombok.Getter;
import lombok.Setter;
import org.json.JSONObject;

import java.util.Objects;

@Getter
@Setter
public class Ingredient {
    private int id;
    private String name;
    private double amount;
    private String unit;

    public void setIngredient(JSONObject json) {
        id = json.getInt("id");
        name = json.getString("name");
        amount = json.getDouble("amount");
        unit = convertUnit(json.getString("unit"));
    }

    // Some units from the external API are incorrect.
    // All found mistakes are corrected here.
    private String convertUnit(String unit) {
        if (Objects.equals(unit, "T") || Objects.equals(unit, "t")) {
            return "tbsp";
        }
        return unit;
    }
}

File: frontend/lib/common/models/ingredient.dart
class Ingredient {
  Ingredient({required this.id, required this.name, required this.amount, required this.unit});
  final int id;
  final String name;
  final double amount;
  final String unit;

  factory Ingredient.fromJson(Map<String, dynamic> json) {
    return Ingredient(
      id: json['id'],
      name: json['name'],
      amount: json['amount'],
      unit: json['unit']
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'amount': amount,
    'unit': unit,
  };
}
